bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/SurePetcareAPIHelper.java

```diff
@@ -0,0 +1,474 @@
+    private static final String TOPOLOGY_URL = API_URL + "/me/start";
+    private static final String PET_BASE_URL = API_URL + "/pet";
+    private static final String PET_STATUS_URL = API_URL + "/pet/?with[]=status&with[]=photo";
+    private static final String DEVICE_BASE_URL = API_URL + "/device";
+    private static final String LOGIN_URL = API_URL + "/auth/login";
+
+    public static final int DEFAULT_DEVICE_ID = 12344711;
+
+    private String authenticationToken = "";
+    private String username = "";
+    private String password = "";
+
+    private @NonNullByDefault({}) HttpClient httpClient;
+    private SurePetcareTopology topologyCache = new SurePetcareTopology();
+
+    /**
+     * Sets the httpClient object to be used for API calls to Sure Petcare.
+     *
+     * @param httpClient the client to be used.
+     */
+    public void setHttpClient(@Nullable HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * This method uses the provided username and password to obtain an authentication token used for subsequent API
+     * calls.
+     *
+     * @param username The Sure Petcare username (email address) to be used
+     * @param password The password
+     * @throws AuthenticationException
+     */
+    public synchronized void login(String username, String password) throws AuthenticationException {
+        try {
+            Request request = httpClient.POST(LOGIN_URL);
+            setConnectionHeaders(request);
+            request.content(new StringContentProvider(SurePetcareConstants.GSON
+                    .toJson(new SurePetcareLoginCredentials(username, password, getDeviceId().toString()))));
+            ContentResponse response = request.send();
+            if (response.getStatus() == HttpURLConnection.HTTP_OK) {
+                SurePetcareLoginResponse loginResponse = SurePetcareConstants.GSON
+                        .fromJson(response.getContentAsString(), SurePetcareLoginResponse.class);
+                if (loginResponse != null) {
+                    authenticationToken = loginResponse.getToken();
+                    this.username = username;
+                    this.password = password;
+                    logger.debug("Login successful, token: {}", authenticationToken);
+                } else {
+                    throw new AuthenticationException("Invalid JSON response from login");
+                }
+            } else {
+                logger.debug("HTTP Response Code: {}", response.getStatus());
+                logger.debug("HTTP Response Msg: {}", response.getReason());
+                throw new AuthenticationException(
+                        "HTTP response " + response.getStatus() + " - " + response.getReason());
+            }
+        } catch (IOException | InterruptedException | TimeoutException | ExecutionException e) {
+            logger.debug("Exception caught during login: {}", e.getMessage());
```

Comment:
Normally you should either log the exception or catch it now you forward it and log it, but given its a debug I would not see this as blocking

Reply:
removed logging as it's already logged by the calling method